### DISCUSSION TITLE
provide clear the task on the other DAG and its downstream tasks recursively without ExternalTaskMarker

### DISCUSSION THREAD

**Discussion by yyhecust** [apache/airflow#40148] [Ideas]
### Description

being able to  clear the task on the other DAG and its downstream tasks recursively without ExternalTaskMarker and don't modify dag files.

### Use case/motivation

In my application, we need to clear the task on the other DAG and its downstream tasks recursively, but when we design the Dag template, we didn't use ExternalTaskMarker and we didn't want to modify the Dag files that were running in production. So, we can't use ExternalTaskMarker to help us clear the task and its downstream tasks over Dag and we want to do this without modifying the dag files.

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by hsnprsd** [comment]
I think it would be a nice feature to have. When defining a sensor user should be able to set whether or not he/she wants the sensor get cleared when upstream task gets into success state. I don't know how to implement it though.

It is also possible to implement a custom solution. You can use airflow api to monitor tasks turning into success state and clear any downstream sensors that they have.
